src/gluonts/evaluation/_base.py

```diff
@@ -185,6 +192,41 @@ def extract_pred_target(
             np.squeeze(time_series.loc[forecast.index].transpose())
         )
 
+    # This method is needed for the owa calculation
+    # It extracts the training sequence from the Series or DataFrame to a numpy array
+    @staticmethod
+    def extract_past_data(
+        time_series: Union[pd.Series, pd.DataFrame], forecast: Forecast
+    ) -> np.ndarray:
+        """
+
+        Parameters
+        ----------
+        time_series
+        forecast
+
+        Returns
+        -------
+        np.ndarray
+            time series without the forecast dates
+        """
+
+        assert forecast.index.intersection(time_series.index).equals(
+            forecast.index
+        ), (
+            "Index of forecast is outside the index of target\n"
+            f"Index of forecast: {forecast.index}\n Index of target: {time_series.index}"
+        )
+
+        # cut the time series using the dates of the forecast object
+        return np.atleast_1d(
+            np.squeeze(
+                time_series.loc[
+                    time_series.index.difference(forecast.index)
```

Comment:
According to pandas’ documentation, `difference` here computed returns 

> a new Index with elements from the index that are not in other

What if the forecast does not fall at the end of the time series, but in the middle of it? The index of the extracted data should just be whatever is before the forecast start date I’d say

Reply:
I agree, great catch!